user-defined-plugin.md

```diff
@@ -0,0 +1,261 @@
+`name` attribute holds Name of the plugin. in this case `riem_measurs`, please make sure to use your directory name for this `name` attribute. 
+
+#### displayName (required)
+
+`displayName` attribute is used for Display Name on Data Source Picker Dialog and Import Data Dialog.
+
+#### iconURL (required)
+
+`iconURL` holds icon image file URL for your plugin. currently it only supports external URL.
+
+For example:
+ 
+```
+   "iconURL" : "http://xmllondon.com/images/sparqlThumb.png",
+```
+
+If you do not have an icon, please use below for default lego like block icon.
+
+```
+   "iconURL" : "lib/images/plugin.png",
+```
+
+If you use external URL, make sure to set
+`iconWidh` and `iconHeight` to 32px to fit icon in the UI.
+
+```
+"iconWidth" : "32px",
+"iconHeight" : "32px"
+```
+
+#### helpURL (required)
+
+`helpURL` holds URL for your plugin help page. Help Link is put on Import Dialog Header.
+if you do not have one, you can set default exploratory doc link like below:
+
+```
+"helpURL" :  "http://docs.exploratory.io/",
+```
+
+#### rPackageDependencies (required)
+
+`rPackageDependencies` is an array of package names that the plugin depends. If your plugin does not depends on any package,
+leave it as empty array like below:
+
+```
+"rPackageDependencies": ["riem"],
+```
+
+#### function (required)
+
+`function` holds R function name that the plugin calls to get a data. The R function must return data frame as output. In this example, you'll call `riem_measures` from riem package.
+
+#### rSourceFile (required)
+
+`rSourceFile` holds a name of R script file that the plugin depends. Let's use `plugin.r` for this example.
+
+#### hasQueryField (optional)
+
+If you want to have a dedicated query input field that has much space for your query string, Set `hasQueryField` as `true` and add a parameter whose `name` is `query` and `dataType` is text.
+
+For example, if you define following in your `plugin.json`
+
+``` 
+"hasQueryField" : true,
+"inputParameters": [
+  {
+    "name" : "query",
+    "displayName":"Query",
+    "dataType":"text",
+    "defaultValue" : "",
+    "required" : true
+  },
+```
+then you can have dedicated query input field on right hand side of the import Dialog.
+
+![](import/images/plugin_with_query_field.png)
+
+### inputParameters
+
+`inputParameters` is an array of parameter passed to the `function` (i.e in this case, `riem_measures`) and these are rendered as input fields on Data Import Dialog. Parameters order matters so make sure to set input parameters in a way that underly R function expects. For example, if your R function has arguments station, start_date, and end_date then you need to define your inputParameters in this order.(i.e station, start_date, and end_date). If you want to define parameter order in a different way, you need to write wrapper function in library.r file and then do a parameter mapping there like below and set the wrapper function name (`riem_measures_wrapper`) to `function` attribute.
```

Comment:
Do we need to talk about library.r here? Shouldn't user define wrapper function in plugin.r?